### PR TITLE
Aegis factions Compatability

### PR DESCRIPTION
**Pull request by RetardedFoX**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Adds Aegis factions (Excluding Argana) to the gamemode.
Mod complains if you run without WS CDLC, but no assets will be used without it enabled (I hope).

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?
3. [x] Hosting locally causes my Windows to combust in on itself.

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Gaming time.
I haven't tested this at all :) It makes my pc shit itself.

********************************************************
Notes:

### CODE REVIEW

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_China.sqf [hunk 1]
```diff
@@ -0,0 +1,938 @@
...
+    ["srifle_DMR_05_ghex_F", "muzzle_snds_93mmg", "acc_pointer_IR", "optic_KHS_blk", [], [], "bipod_02_F_blk"],
+    ["srifle_DMR_05_blk_F", "muzzle_snds_93mmg", "acc_pointer_IR", "optic_LRPS", [], [], "bipod_02_F_blk"]
+    ];
+    
+    (_militaryLoadoutData get "machineGuns") append [
+    ["MMG_01_black_F", "", "acc_pointer_IR", "optic_Arco", [], [], "bipod_02_F_blk"],
+    ["MMG_01_ghex_F", "", "acc_pointer_IR", "optic_MRCO", [], [], "bipod_02_F_blk"]
+	];
+	(_militaryLoadoutData get "shotGuns") append [
+	["sgun_aa40_lxWS", "", "acc_pointer_IR", "optic_Holosight_smg_blk_F", ["8Rnd_12Gauge_AA40_Pellets_lxWS", "8Rnd_12Gauge_AA40_Slug_lxWS"], [], ""],
```
Comment (ante185) [inline A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_China.sqf:423]: this should prob not be dependent on the marksmen DLC

**Review by ante185 — COMMENTED** [review]
Could perhaps use more Aegis stuff in place of vanilla weapons

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_China.sqf (hunk 1 above)
Comment (RetardedFoX) [reply A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_China.sqf]: Fixed in 360c426

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_BAF_Arid.sqf [hunk 2]
```diff
@@ -0,0 +1,923 @@
...
+["vehiclesTanks", ["B_MBT_01_TUSK_F", "B_MBT_01_cannon_F"]] call _fnc_saveToTemplate;
+["vehiclesAA", ["B_APC_Tracked_01_AA_F"]] call _fnc_saveToTemplate;
+
+["vehiclesTransportBoats", ["B_Boat_Transport_01_F"]] call _fnc_saveToTemplate;
+["vehiclesGunBoats", ["B_Boat_Armed_01_minigun_F"]] call _fnc_saveToTemplate;
+["vehiclesAmphibious", []] call _fnc_saveToTemplate;
+
+["vehiclesPlanesCAS", ["B_A_Plane_Fighter_05_Cluster_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesAA", ["B_A_Plane_Fighter_05_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesTransport", ["B_A_VTOL_01_infantry_F"]] call _fnc_saveToTemplate;
```
Comment (ante185) [inline A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_BAF_Arid.sqf:41]:
```suggestion
["vehiclesPlanesTransport", ["B_A_Plane_Transport_01_infantry_F"]] call _fnc_saveToTemplate;
```

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_China.sqf [hunk 3]
```diff
@@ -0,0 +1,950 @@
...
+private _Tanks = ["O_T_MBT_02_cannon_ghex_F"];
+["vehiclesAA", ["O_T_APC_Tracked_02_AA_ghex_F"]] call _fnc_saveToTemplate;
+
+["vehiclesTransportBoats", ["O_T_Boat_Transport_01_F"]] call _fnc_saveToTemplate;
+["vehiclesGunBoats", ["O_T_Boat_Armed_01_hmg_F"]] call _fnc_saveToTemplate;
+["vehiclesAmphibious", ["O_T_APC_Wheeled_02_rcws_v2_ghex_F"]] call _fnc_saveToTemplate;
+
+["vehiclesPlanesCAS", ["O_T_Plane_CAS_02_dynamicLoadout_ghex_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesAA", ["O_T_Plane_Fighter_02_ghex_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesTransport", ["O_T_VTOL_02_infantry_dynamicLoadout_F", "O_T_Plane_Transport_01_infantry_ghex_F"]] call _fnc_saveToTemplate;
```
Comment (ante185) [inline A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_China.sqf:41]:
```suggestion
["vehiclesPlanesTransport", ["O_T_Plane_Transport_01_infantry_ghex_F"]] call _fnc_saveToTemplate;
```

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_LDF.sqf [hunk 4]
```diff
@@ -0,0 +1,924 @@
...
+["vehiclesTanks", ["Aegis_I_E_MBT_03_cannon_F"]] call _fnc_saveToTemplate;
+["vehiclesAA", ["B_W_APC_Tracked_01_AA_F"]] call _fnc_saveToTemplate;
+
+["vehiclesTransportBoats", ["I_Boat_Transport_01_F"]] call _fnc_saveToTemplate;
+["vehiclesGunBoats", ["I_Boat_Armed_01_minigun_F"]] call _fnc_saveToTemplate;
+["vehiclesAmphibious", ["Aegis_I_E_APC_Wheeled_01_cannon_v2_F"]] call _fnc_saveToTemplate;
+
+["vehiclesPlanesCAS", ["I_E_Plane_Fighter_04_Cluster_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesAA", ["I_E_Plane_Fighter_04_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesTransport", []] call _fnc_saveToTemplate;
```
Comment (ante185) [inline A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_LDF.sqf:41]:
```suggestion
["vehiclesPlanesTransport", ["B_T_Plane_Transport_01_infantry_F"]] call _fnc_saveToTemplate;
```

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_US_Arid.sqf [hunk 5]
```diff
@@ -0,0 +1,1038 @@
...
+private _Tanks = ["B_MBT_01_TUSK_F", "B_MBT_01_cannon_F"];
+["vehiclesAA", ["B_APC_Tracked_01_AA_F"]] call _fnc_saveToTemplate;
+
+["vehiclesTransportBoats", ["B_Boat_Transport_01_F"]] call _fnc_saveToTemplate;
+["vehiclesGunBoats", ["B_Boat_Armed_01_minigun_F"]] call _fnc_saveToTemplate;
+["vehiclesAmphibious", ["a3a_B_APC_Wheeled_01_cannon_F"]] call _fnc_saveToTemplate;
+
+["vehiclesPlanesCAS", ["B_Plane_CAS_01_dynamicLoadout_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesAA", ["B_Plane_Fighter_01_F", "B_Plane_Fighter_05_F"]] call _fnc_saveToTemplate;
+["vehiclesPlanesTransport", ["B_T_VTOL_01_infantry_F","B_T_VTOL_01_infantry_blue_F"]] call _fnc_saveToTemplate;
```
Comment (ante185) [inline A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_US_Arid.sqf:41]:
```suggestion
["vehiclesPlanesTransport", ["B_T_Plane_Transport_01_infantry_F"]] call _fnc_saveToTemplate;
```

**Review by ante185 — COMMENTED** [review]
Suggested changes

**Comment by ante185** [comment]
Any new vehiclesPlanesCAS and vehiclesPlanesAA classnames are going to need a class each for their loadouts in core/Templates/AircraftLoadouts/Aegis, even if they're just inheriting from the existing vanilla loadouts.

Each faction in the template.hpp are going to need an lore entry, see @Tiny-DM for instructions.

File: A3A/addons/core/Templates/AircraftLoadouts/Aegis/config.cpp [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    };
+};
+
+class A3A {
+    class Loadouts
+    {
+        class CASPlane
+        {
+            class baseCAS;
+            class B_Plane_CAS_01_dynamicLoadout_F : baseCAS {
```
Comment (ante185) [inline A3A/addons/core/Templates/AircraftLoadouts/Aegis/config.cpp:24]: Do not re-specify the vanilla loadouts

Comment (RetardedFoX) [reply A3A/addons/core/Templates/AircraftLoadouts/Aegis/config.cpp:24]: Woopsie

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_LDF.sqf [hunk 7]
```diff
@@ -0,0 +1,978 @@
...
+]];
+_militiaLoadoutData set ["grenadeLaunchers", [
+["Aegis_arifle_AK74_GL_F", "", "acc_flashlight_pistol", "", ["30Rnd_545x39_Mag_F", "30Rnd_545x39_Mag_F", "30Rnd_545x39_Mag_Tracer_F"], ["1Rnd_HE_Grenade_shell", "1Rnd_HE_Grenade_shell", "1Rnd_Smoke_Grenade_shell"], ""],
+["Aegis_arifle_AK74_GL_oak_F", "", "acc_flashlight_pistol", "", ["30Rnd_545x39_Mag_F", "30Rnd_545x39_Mag_F", "30Rnd_545x39_Mag_Tracer_F"], ["1Rnd_HE_Grenade_shell", "1Rnd_HE_Grenade_shell", "1Rnd_Smoke_Grenade_shell"], ""]
+]];
+_militiaLoadoutData set ["SMGs", [
+["SMG_03C_black", "", "", "", [], [], ""]
+]];
+_militiaLoadoutData set ["machineGuns", [
+["arifle_RPK_F", "", "acc_flashlight_pistol", "", ["75Rnd_762x39_Mag_F", "75Rnd_762x39_Mag_F", "75Rnd_762x39_Mag_Tracer_F"], [], ""]
```
Comment (ante185) [inline A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_LDF.sqf:383]: Consider giving military and especially militia units less high capacity mags when those mags are compatible with the regular rifles.

Comment (ante185) [inline A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_LDF.sqf:383]: You wanted a reminder that there exists a 5.45 RPK with wooden stock

**Review by ante185 — CHANGES_REQUESTED** [review]
Look over the traitor loadout data, they don't seem to align with expectations.
the rebel factions don't need to start with SMGs if you're giving them semi automatic shotguns.
Situationally you should perhaps use a couple of the guns the military uses in the SF category but otherwise the SF stuff looks good.

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_BAF_Tropical.sqf [hunk 8]
```diff
@@ -0,0 +1,942 @@
...
+_loadoutData set ["items_unarmed_extras", []];
+
+//TODO - ACE overrides for misc essentials, medical and engineer gear
+
+///////////////////////////////////////
+//    Special Forces Loadout Data    //
+///////////////////////////////////////
+
+private _sfLoadoutData = _loadoutData call _fnc_copyLoadoutData; // touch and shit breaks
+_sfLoadoutData set ["uniforms", ["U_B_UBACS_mtp_f", "U_B_UBACS_vest_tna_f", "U_B_UBACS_tshirt_tna_f"]];
```
[the remark this answers is not in the document]
Comment (RetardedFoX) [reply A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_BAF_Tropical.sqf]: 7222e5c

File: A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_BAF_Temperate.sqf [hunk 9]
```diff
@@ -0,0 +1,975 @@
...
+["SMG_04_blk_F", "", "", "optic_Aco_smg", [], [], ""],
+["SMG_04_blk_F", "", "", "Aegis_optic_ROS_SMG", [], [], ""]
+]];
+_militaryLoadoutData set ["machineGuns", [
+["LMG_Mk200_black_F", "", "acc_pointer_IR", "optic_Holosight_blk_F", ["200Rnd_65x39_cased_Box","200Rnd_65x39_cased_Box","200Rnd_65x39_cased_Box_Tracer_Red"], [], "bipod_01_F_blk"],
+["LMG_Mk200_black_F", "", "acc_pointer_IR", "optic_ACO_grn", ["200Rnd_65x39_cased_Box","200Rnd_65x39_cased_Box","200Rnd_65x39_cased_Box_Tracer_Red"], [], "bipod_01_F_blk"],
+["LMG_Mk200_black_F", "", "acc_pointer_IR", "Aegis_optic_ROS", ["200Rnd_65x39_cased_Box","200Rnd_65x39_cased_Box","200Rnd_65x39_cased_Box_Tracer_Red"], [], "bipod_01_F_blk"]
+]];
+_militaryLoadoutData set ["marksmanRifles", [
+["arifle_MXM_Black_F", "", "acc_pointer_IR", "optic_SOS", ["30Rnd_65x39_caseless_black_mag", "30Rnd_65x39_caseless_black_mag", "30Rnd_65x39_caseless_black_mag_Tracer"], [], "bipod_01_F_blk"],
```
[the remark this answers is not in the document]
Comment (RetardedFoX) [reply A3A/addons/core/Templates/Templates/Aegis/Aegis_AI_BAF_Temperate.sqf]: 7222e5c

**Review by ante185 — APPROVED** [review]
Good for testing

**Pull request merged**